whelk-core/src/main/groovy/whelk/component/ElasticClient.groovy

```diff
@@ -52,50 +52,63 @@ class ElasticClient {
+            HttpConnectionParams.setConnectionTimeout(httpParams, 0)
+            HttpConnectionParams.setSoTimeout(httpParams, 0)
+        }
         // FIXME: upgrade httpClient (and use RequestConfig)- https://issues.apache.org/jira/browse/HTTPCLIENT-1418
         // httpParams.setParameter(ClientPNames.CONN_MANAGER_TIMEOUT, new Long(TIMEOUT_MS));
 
-        return new ElasticClient(httpClient, elasticHosts)
+        return new ElasticClient(httpClient, elasticHosts, useTimeouts)
     }
 
-    ElasticClient(HttpClient httpClient, List<String> elasticHosts) {
+    ElasticClient(HttpClient httpClient, List<String> elasticHosts, boolean useTimeouts) {
         this.httpClient = httpClient
         this.elasticNodes = elasticHosts.collect { new ElasticNode(it) }
+        this.useTimeouts = useTimeouts;
 
-        globalRetry = retryRegistry.retry(ElasticClient.class.getSimpleName(), RetryConfig.custom()
-                .waitDuration(Duration.ofMillis(10))
-                .maxAttempts(elasticNodes.size() * 2)
-                .build())
+        if (useTimeouts) {
+            globalRetry = retryRegistry.retry(ElasticClient.class.getSimpleName(), RetryConfig.custom()
+                    .waitDuration(Duration.ofMillis(10))
+                    .maxAttempts(elasticNodes.size() * 2)
+                    .build())
 
-        CollectorRegistry collectorRegistry = CollectorRegistry.defaultRegistry
-        collectorRegistry.register(RetryMetricsCollector.ofRetryRegistry(retryRegistry))
-        collectorRegistry.register(CircuitBreakerMetricsCollector.ofCircuitBreakerRegistry(circuitBreakerRegistry))
+            CollectorRegistry collectorRegistry = CollectorRegistry.defaultRegistry
+            collectorRegistry.register(RetryMetricsCollector.ofRetryRegistry(retryRegistry))
+            collectorRegistry.register(CircuitBreakerMetricsCollector.ofCircuitBreakerRegistry(circuitBreakerRegistry))
+        }
 
         log.info "ElasticSearch component initialized with ${elasticHosts.size()} nodes."
     }
 
     Tuple2<Integer, String> performRequest(String method, String path, String body, String contentType0 = null) {
-        try {
-            def nodes = cycleNodes()
-            return globalRetry.executeSupplier({ -> nodes.next().performRequest(method, path, body, contentType0) })
-        }
-        catch (Exception e) {
-            log.warn("Request to ElasticSearch failed: ${e}", e)
-            throw new ElasticIOException(e.getMessage(), e)
+        if (useTimeouts) {
+            try {
+                def nodes = cycleNodes()
+                return globalRetry.executeSupplier({ -> nodes.next().performRequest(method, path, body, contentType0) })
+            }
+            catch (Exception e) {
+                log.warn("Request to ElasticSearch failed: ${e}", e)
+                throw new ElasticIOException(e.getMessage(), e)
+            }
+        } else {
+            elasticNodes[0].performRequest(method, path, body, contentType0)
```

Comment:
Should use all elastic nodes.
(e.g. `elasticNodes.get(random.nextInt(elasticNodes.size()))`)

Reply:
Elastic rekommenderar att man indexerar till en nod, inte alla. Åtminstonde är det min tolkning av detta:
https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-indexing-speed.html#_disable_refresh_and_replicas_for_initial_loads

Reply:
Jag tolkar det som att man ska stänga av replikering mellan elastic-noderna (vilket vi borde göra). Normalt finns det en replika av varje shard.

dvs:
Innan reindex
```
PUT /<index>/_settings
{
    "index" : {
        "number_of_replicas" : 0
        "refresh_interval" : -1
    }
}
```

Efter reindex
```
PUT /<index>/_settings
{
    "index" : {
        "number_of_replicas" : 1
        "refresh_interval" : 1
    }
}
```